src/wallet/main.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::ffi::OsString;
+
+use super::subcommands::Cli;
+use crate::networks::NetworkChain;
+use crate::rpc_client::chain_get_name;
+use crate::shim::address::{CurrentNetwork, Network};
+use clap::Parser;
+use std::str::FromStr;
+
+pub fn main<ArgT>(args: impl IntoIterator<Item = ArgT>) -> anyhow::Result<()>
+where
+    ArgT: Into<OsString> + Clone,
+{
+    // Capture Cli inputs
+    let Cli { opts, cmd } = Cli::parse_from(args);
```

Comment:
The command-line options are not appropriate for the wallet tool.

Reply:
Oh, I see, this `Cli` is different from the other `Cli` structures. All good, then.